### PR TITLE
feat(datastore): do not compress write ahead log file in case compression fails

### PR DESCRIPTION
**Pull request by xuchuan**
…sion fails

## Modules
- [ ] UI
- [X] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [X] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java [hunk 1]
```diff
@@ -143,18 +152,34 @@ private void getNext() {
...
+                }
                 int uncompressedSize;
-                try {
-                    var inBuf = data.asByteBuffer();
-                    var outBuf = this.buf.asByteBuffer();
-                    if (inBuf.hasArray()) {
-                        uncompressedSize = Snappy.uncompress(inBuf.array(), 0, inBuf.capacity(), outBuf.array(), 0);
-                    } else {
-                        uncompressedSize = Snappy.uncompress(inBuf, outBuf);
+                var h = new byte[4];
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java:160]: 4 -> `WalManager.this.header.length` ?

File: server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java [hunk 2]
```diff
@@ -275,25 +300,35 @@ private void writeToObjectStore(boolean clearOutput) {
...
                 inBuf.limit(this.outputStream.getOffset());
                 compressedSize = Snappy.compress(inBuf, outBuf);
             }
         } catch (IOException e) {
-            log.error("data loss: failed to compress", e);
-            return;
+            log.warn("failed to compress", e);
+            this.header[3] = WalManager.NO_COMPRESSION;
+            this.compressedBuffer.setBytes(0, this.header, 0, this.header.length);
+            this.outputBuffer.copyTo(this.compressedBuffer.slice(4, this.outputStream.getOffset()));
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/datastore/WalManager.java:319]: `this.header.length` ?

**Comment by jialeicui** [comment]
LGTM

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**